app/src/main/java/io/appium/settings/RecorderThread.java

```diff
@@ -0,0 +1,437 @@
+/*
+  Copyright 2012-present Appium Committers
+  <p>
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package io.appium.settings;
+
+import android.hardware.display.DisplayManager;
+import android.hardware.display.VirtualDisplay;
+import android.media.AudioAttributes;
+import android.media.AudioFormat;
+import android.media.AudioPlaybackCaptureConfiguration;
+import android.media.AudioRecord;
+import android.media.MediaCodec;
+import android.media.MediaCodecInfo;
+import android.media.MediaFormat;
+import android.media.MediaMuxer;
+import android.media.projection.MediaProjection;
+import android.os.Build;
+import android.os.Handler;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.Surface;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import androidx.annotation.RequiresApi;
+
+import static io.appium.settings.RecorderConstant.NO_TIMESTAMP_SET;
+
+public class RecorderThread implements Runnable {
+
+    private static final String TAG = "RecorderThread";
+
+    private final MediaProjection mediaProjection;
+    private final String outputFilePath;
+    private final int videoWidth;
+    private final int videoHeight;
+    private final int recordingRotation;
+
+    private boolean muxerStarted;
+    private boolean isStartTimestampInitialized = false;
+    private long startTimestampUs;
+    private long lastAudioTimestampUs = NO_TIMESTAMP_SET;
+
+    private int videoTrackIndex = -1;
```

Comment:
It is better readable when constants are used instead of magic numbers

Reply:
done